sdcm/nemesis.py

```diff
@@ -2347,6 +2357,119 @@ def modify_table_speculative_retry(self):
+
+                example of settings param:
+                    {"name": ks_cf,
+                     "compaction":  {'class': 'TimeWindowCompactionStrategy',
+                                     'compaction_window_size': '1',
+                                     'compaction_window_unit': 'MINUTES'},
+                     "gc": 864000,
+                     "dttl": 86400}
+
+            """
+            current_unit = settings["compaction"]["compaction_window_unit"]
+            current_size = int(settings["compaction"]["compaction_window_size"])
+            multiplier = 3600 if current_unit in ["DAYS", "HOURS"] else 60
+            expected_sstable_number = 35
+
+            if current_unit == "DAYS":
+                current_size = current_size + 1
+            elif current_unit == "HOURS":
+                if (current_size // 24) > 2:
+                    current_unit = "DAYS"
+                    current_size = 3
+                else:
+                    current_size += 10
+            else:
+                if (current_size // 60) > 10:
+                    current_unit = "HOURS"
+                    current_size = 11
+                else:
+                    current_size += 35
+
+            settings["gc"] = current_size * multiplier * expected_sstable_number // 2
+            settings["dttl"] = current_size * multiplier * expected_sstable_number
+            settings["compaction"]["compaction_window_unit"] = current_unit
+            settings["compaction"]["compaction_window_size"] = current_size
+
+            return settings
+
+        all_ks_cs_with_twcs = self.cluster.get_all_tables_with_twcs(self.target_node)
+        self.log.debug("All tables with TWCS %s", all_ks_cs_with_twcs)
+
+        if not all_ks_cs_with_twcs:
+            raise UnsupportedNemesis('No table found with TWCS')
+
+        target_ks_cs_with_settings = random.choice(all_ks_cs_with_twcs)
+
+        ks_cs_settings = set_new_twcs_settings(target_ks_cs_with_settings)
+        keyspace, table = ks_cs_settings["name"].split(".")
+
+        num_sstables_before_change = len(self.target_node.get_list_of_sstables(keyspace, table, suffix="-Data.db"))
+
+        self.log.debug("New TWCS settings: %s", str(ks_cs_settings))
+        self._modify_table_property(
+            name="compaction", val=ks_cs_settings["compaction"], keyspace_table=ks_cs_settings["name"])
+        self._modify_table_property(name="default_time_to_live",
+                                    val=ks_cs_settings["dttl"], keyspace_table=ks_cs_settings["name"])
+        self._modify_table_property(name="gc_grace_seconds",
+                                    val=ks_cs_settings["gc"], keyspace_table=ks_cs_settings["name"])
+
+        self.cluster.wait_for_schema_agreement()
+        # wait timeout  equal 2% of test duration for generating sstables with timewindow settings
+        sleep_timeout = int(0.02 * self.tester.params["test_duration"])
```

Comment:
@aleksbykov can you explain again this calculation ?

test_duration is in minutes, while `sleep_timeout` is in seconds

so for a 3 day test:
4320 * 0.02 = 86.4

for a 4h test:
360 * 0.02 = 7.2

so for one test it would sleep for a bit more then minute, and for the other it would sleep 7 sec, I fail to understand how the duration play any part of the TWCS configuration, and why the timing here isn't base on the selected table configuration...